packages/cw721-controllers/src/nft_claim.rs

```diff
@@ -0,0 +1,422 @@
+use cosmwasm_std::{Addr, BlockInfo, CustomQuery, Deps, StdResult, Storage};
+use cw_storage_plus::Map;
+use cw_utils::Expiration;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct NftClaimsResponse {
+    pub nft_claims: Vec<NftClaim>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct NftClaim {
+    pub token_id: String,
+    pub release_at: Expiration,
+}
+
+impl NftClaim {
+    pub fn new(token_id: String, released: Expiration) -> Self {
+        NftClaim {
+            token_id,
+            release_at: released,
+        }
+    }
+}
+
+pub struct NftClaims<'a>(Map<'a, &'a Addr, Vec<NftClaim>>);
+
+impl<'a> NftClaims<'a> {
+    pub const fn new(storage_key: &'a str) -> Self {
+        NftClaims(Map::new(storage_key))
+    }
+
+    /// This creates an Nft claim, such that the given address can claim an Nft after
+    /// the release date.
+    pub fn create_nft_claim(
+        &self,
+        storage: &mut dyn Storage,
+        addr: &Addr,
+        token_id: String,
+        release_at: Expiration,
+    ) -> StdResult<()> {
+        // add a claim to this user to get their tokens after the unbonding period
+        self.0.update(storage, addr, |old| -> StdResult<_> {
+            let mut nft_claims = old.unwrap_or_default();
+            nft_claims.push(NftClaim {
+                token_id,
+                release_at,
+            });
+            Ok(nft_claims)
+        })?;
+        Ok(())
+    }
+
+    /// Creates a number of NFT claims simeltaniously for a given
+    /// address. Similar to create_nft_claim but does only one storage
+    /// read and write for multiple claims.
+    pub fn create_nft_claims(
```

Comment:
can we just always use this function instead of create_nft_claim, and in the case of a single token_id, have a vector with a single element?

Reply:
Great idea. We don't even use the non vec version at the moment.